### PR TITLE
Added machine annotation service

### PR DESCRIPTION
**Pull request by samleeflang**
Added Machine Annotation Service.
Some minor changes to the rest:
- formatting
- fixes in controller where we did not follow JSON:API
- minor optimization to tests

https://naturalis.atlassian.net/browse/DD-539

### CODE REVIEW

**Comment by samleeflang** [comment]
Sonar fails because of condition coverage. This is due to changes in the HandleService which doesn't have tests and will be replaced soon. So I want to leave that be and don't fix the Sonar coverage

File: src/main/java/eu/dissco/orchestration/backend/controller/SourceSystemController.java [hunk 1]
```diff
@@ -58,9 +58,13 @@ public ResponseEntity<JsonApiWrapper> updateSourceSystem(Authentication authenti
...
     var id = prefix + '/' + suffix;
     log.info("Received update request for source system: {}", id);
-    String path = SANDBOX_URI + servletRequest.getRequestURI();
+    String path = appProperties.getBaseUrl() + servletRequest.getRequestURI();
     var result = service.updateSourceSystem(id, sourceSystem, path);
-    return ResponseEntity.ok(result);
+    if (result == null) {
+      return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+    } else {
+      return ResponseEntity.ok(result);
```
Comment (southeo) [inline src/main/java/eu/dissco/orchestration/backend/controller/SourceSystemController.java:66]: can take this out of the else block too

File: src/main/java/eu/dissco/orchestration/backend/properties/ApplicationProperties.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.validation.annotation.Validated;
+
+@Data
+@Validated
+@Component
+@ConfigurationProperties("application")
+public class ApplicationProperties {
+
+  @NotBlank
+  private String baseUrl;
```
Comment (southeo) [inline src/main/java/eu/dissco/orchestration/backend/properties/ApplicationProperties.java:17]: Where do you set this value?

File: src/main/java/eu/dissco/orchestration/backend/repository/MachineAnnotationServiceRepository.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+@Repository
+@RequiredArgsConstructor
+public class MachineAnnotationServiceRepository {
+
+  private final DSLContext context;
+  private final ObjectMapper mapper;
+
+  public void createMachineAnnotationService(MachineAnnotationServiceRecord masRecord) {
+    context.insertInto(MACHINE_ANNOTATION_SERVICES)
```
Comment (southeo) [inline src/main/java/eu/dissco/orchestration/backend/repository/MachineAnnotationServiceRepository.java:30]: If these properties are set to null, you'd push a null value in the repository. Is that the behaviour you want?

File: src/test/java/eu/dissco/orchestration/backend/controller/MachineAnnotationServiceControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    assertThat(result.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+  }
+
+  @Test
+  void testCreateMappingBadType() {
+    // Given
+    var requestBody = givenSourceSystemRequest();
+
+    // Then
+    assertThrowsExactly(
```
Comment (southeo) [inline src/test/java/eu/dissco/orchestration/backend/controller/MachineAnnotationServiceControllerTest.java:64]: What's the difference between assertThrows and assertThrowsExactly?

File: src/test/resources/db/migration/V1__init_db.sql [hunk 5]
```diff
@@ -21,4 +21,27 @@ CREATE TABLE public.new_mapping (
 	deleted timestamptz NULL,
 	sourcedatastandard varchar NOT NULL,
 	CONSTRAINT new_mapping_pk PRIMARY KEY (id, version)
+);
+
+create table machine_annotation_services
+(
+    id                            text                     not null
+        primary key,
```
Comment (southeo) [inline src/test/resources/db/migration/V1__init_db.sql:29]: fix formatting

**Review by southeo — COMMENTED** [review]
Looks good, few clarifying questions. 

Funny how we have machineAnnotationServiceService.

File: src/main/java/eu/dissco/orchestration/backend/controller/MappingController.java [hunk 6]
```diff
@@ -37,38 +38,44 @@
...
     var mapping = getMappingFromRequest(requestBody);
     var id = prefix + '/' + suffix;
     log.info("Received update request for mapping: {}", id);
-    String path = SANDBOX_URI + request.getRequestURI();
+    String path = appProperties.getBaseUrl() + servletRequest.getRequestURI();
     var result = service.updateMapping(id, mapping, getNameFromToken(authentication), path);
-    return ResponseEntity.ok(result);
+    if (result == null) {
+      return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+    } else {
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/orchestration/backend/controller/MappingController.java:69]: hmm I like it in an `else` block. It makes the distinction clear, plus if we ever change anything it won't automatically become the `.ok(result)`.

File: src/main/java/eu/dissco/orchestration/backend/controller/SourceSystemController.java (hunk 1 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/orchestration/backend/controller/SourceSystemController.java:66]: hmm I like it in an `else` block. It makes the distinction clear, plus if we ever change anything it won't automatically become the `.ok(result)`.

File: src/main/java/eu/dissco/orchestration/backend/properties/ApplicationProperties.java (hunk 2 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/orchestration/backend/properties/ApplicationProperties.java:17]: In the properties file and when deploying in the env variables file

File: src/main/java/eu/dissco/orchestration/backend/repository/MachineAnnotationServiceRepository.java (hunk 3 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/orchestration/backend/repository/MachineAnnotationServiceRepository.java:30]: Yes some values can be null, they are not really required.

File: src/test/java/eu/dissco/orchestration/backend/controller/MachineAnnotationServiceControllerTest.java (hunk 4 above)
Comment (samleeflang) [reply src/test/java/eu/dissco/orchestration/backend/controller/MachineAnnotationServiceControllerTest.java:64]: assertThrows will also pass if it a child object of the Exception

File: src/main/java/eu/dissco/orchestration/backend/controller/MappingController.java [hunk 7]
```diff
@@ -37,38 +38,44 @@
 public class MappingController {
 
   private final MappingService service;
-  private static final String SANDBOX_URI = "https://sandbox.dissco.tech/orchestrator";
   private final ObjectMapper mapper;
-  private static final ArrayList<String> sourceDataSystems = new ArrayList<>(List.of("dwc", "abcd", "abcdefg"));
+  private final ApplicationProperties appProperties;
+  private static final ArrayList<String> sourceDataSystems = new ArrayList<>(
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/orchestration/backend/controller/MappingController.java:43]: :+1:

File: src/main/java/eu/dissco/orchestration/backend/controller/MappingController.java [hunk 8]
```diff
@@ -77,10 +84,10 @@ public ResponseEntity<Void> deleteMapping(
   @ResponseStatus(HttpStatus.OK)
   @GetMapping(value = "/{prefix}/{postfix}", produces = MediaType.APPLICATION_JSON_VALUE)
   public ResponseEntity<JsonApiWrapper> getMappingById(@PathVariable("prefix") String prefix,
-      @PathVariable("postfix") String postfix, HttpServletRequest request) {
+      @PathVariable("postfix") String postfix, HttpServletRequest servletRequest) {
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/orchestration/backend/controller/MappingController.java:87]: Oh sorry forgot this one

[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/orchestration/backend/controller/MappingController.java:87]: :+1:

**Pull request merged**